scripts/updateDatabase.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+## This script will create or replace a database using a mysql dump file. It takes the DB name (-d or --database) and dump filepath (-f or --file) as arguments.
+## A config file populated with mySQL username, password and host is also required. Its filepath can be optionally specified (-c or --config), otherwise a default 'config.properties' file is assumed.
+## author: jcook
+
+dbFilepath=
+dbName=
+configFilepath=config.properties
+## Parse command-line arguments
+while (( "$#" )); do
+	case "$1" in
+		-f|--file)
+			dbFilepath=$2;
+			shift 2;
+			;;
+		-d|--database)
+			dbName=$2;
+			shift 2;
+			;;
+		-c|--config)
+			configFilepath=$2;
+			shift 2;
+			;;
+		-*|--*=)
+			echo "Error: Unsupported flag $1"
+			exit 1
+	esac
+done
+
+## If missing arguments, explain usage
+if [ -z "$dbFilepath" ] || [ -z "$dbName" ]
+then
+	echo "Create or replace mySQL databases";
+	echo "Usage: bash updateDatabase.sh -f databaseFile -d databaseName [-c configFile] ";
+	exit 1
+fi
+
+## Parse config file for username and password properties
+echo "Reading $configFilepath";
+username=
+password=
+host=localhost
+port=3306
+while read line; do
+	if [[ $line == username* ]]
+ 	then
+ 		username=${line#*=}
+ 	elif [[ $line == password* ]]
+ 	then
+ 		password=${line#*=}
+	elif [[ $line == host* ]]
+	then
+		host=${line#*=}
+	elif [[ $line == port* ]]
+	then
+		port=${line#*=}
+ 	fi
+done < $configFilepath
+
+## Improperly formatted or missing config file
+if [ -z "$username" ] || [ -z "$password" ]
+then
+	echo "Could not find username and/or password in $configFilepath";
+	exit 1
+fi
+
+echo "Updating $dbName with $dbFilepath";
+echo "Backing up $dbName";
+
+## Take archive of DB, drop it and create a new, empty one
+dbArchiveFile="$dbName.backup.dump"
+mysqldump -u$username -p$password -h$host -P$port $dbName > $dbArchiveFile
```

Comment:
I'm not sure if it's doable, but is there a way to do the mysqldump without specifying the password as an argument?  It's potentially visible to users of the system while it's running.

Reply:
Yea we had talked about doing this. I'll try and find a better way of doing this before I merge

Reply:
I think you can specify a my.cnf file containing the username and password to the mysqldump command that it will use to parse the credentials.  The updateDatabase.sh script could take the path to the my.cnf file as an argument.

https://stackoverflow.com/a/24369804

Reply:
Done